### DISCUSSION TITLE
[ZK Stack] - <about zksync-state-reconstruct usage>

### DISCUSSION THREAD

**Discussion by qiwenzhou** [zkSync-Community-Hub/zksync-developers#609] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I accidently deleted my local db and lost server data of my personal hyperchain. For coins that stuck in this dev env, I want to reconstruct this env by zksync-state-reconstruct tool. Is it possible to restart server by reconstruct data from L1? 
If possible, how can i achieve the goal? How to set  zksync-state-reconstruct for restore my hyperchain data?

### Expected Behavior

reconstruct env

### Current Behavior

lost my L2 data.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by bxpana** [comment]
Will check on this for you.

**Reply by bxpana** [reply, marked as answer]
It's not fully supported, but we are finalizing this feature and will let you know when it's released.

**Reply by qiwenzhou** [reply]
Thanks for that. I wonder that is there a solution that I can withdraw my eth from bridge as I understand all coin deposited is actually locked in bridge?

**Reply by qiwenzhou** [reply]
Really need help here！

**Reply by bxpana** [reply]
You would need a fully functioning L2 node in order to withdraw funds which would mean restoring Postgres data
